### PR TITLE
Remove the LockedMod and LockedMods types from optimiser.

### PR DESCRIPTION
**Pull request by ryan-rushton**
The majority of this is just removing the types and fixing the code as needed. 

The mod state in the optimiser is now an array of PluggableInventoryItemDefs rather than the map of plug category hashes to arrays.

### CODE REVIEW

File: src/app/loadout-builder/filter/ModPicker.tsx
Comment (ryan-rushton) [inline src/app/loadout-builder/filter/ModPicker.tsx:251]: I got rid of the boolean condition here as the footer popping in and out when choosing the first mod has always annoyed me.

File: src/app/loadout-builder/mod-utils.ts
Comment (ryan-rushton) [inline src/app/loadout-builder/mod-utils.ts:152]: Added some extra sorting conditions here so that mod order matches the order of the mod picker sections. Note that it kinda matches the logic in sortModGroups.

File: src/app/loadout-builder/filter/LockArmorAndPerks.tsx
Comment (nev-r) [inline src/app/loadout-builder/filter/LockArmorAndPerks.tsx:183]: should never use array index for key, you will be *shocked* at the wild cases where it wrongly decides elements can be re-used

File: src/app/loadout-builder/generated-sets/CompareDrawer.tsx
Comment (nev-r) [inline src/app/loadout-builder/generated-sets/CompareDrawer.tsx:255]: here too

**Review by nev-r — APPROVED** [review]
LO behaviors seeeeeeeeem unchanged from my spot testing

File: src/app/loadout-builder/filter/LockArmorAndPerks.tsx
Comment (ryan-rushton) [reply src/app/loadout-builder/filter/LockArmorAndPerks.tsx]: forgot to go back and fix this, done

File: src/app/loadout-builder/generated-sets/CompareDrawer.tsx
Comment (ryan-rushton) [reply src/app/loadout-builder/generated-sets/CompareDrawer.tsx]: done

**Comment by ryan-rushton** [comment]
FYI I am going to wait until after release to merge this in.

**Pull request merged**
